rust_snuba/src/consumer.rs

```diff
@@ -12,6 +25,58 @@ struct Args {
     settings_path: String,
 }
 
+struct Noop {}
+impl<T> ProcessingStrategy<T> for Noop where T: Clone {
+    fn poll(&mut self) -> Option<CommitRequest> {
+        None
+    }
+    fn submit(&mut self, _message: Message<T>) -> Result<(), MessageRejected> {
+        Ok(())
+    }
+    fn close(&mut self) {}
+    fn terminate(&mut self) {}
+    fn join(&mut self, _timeout: Option<Duration>) -> Option<CommitRequest> {
+        None
+    }
+}
+
+#[derive(Clone)]
+struct ClickhouseInsert {}
+
+struct PythonTransformStep {
+    pub next_step: Box<dyn ProcessingStrategy<ClickhouseInsert>>,
+}
+
+impl ProcessingStrategy<KafkaPayload> for PythonTransformStep {
+    fn poll(&mut self) -> Option<CommitRequest> {
+        self.next_step.poll()
+    }
+
+    fn submit(&mut self, message: Message<KafkaPayload>) -> Result<(), MessageRejected> {
+        log::info!("processing message");
```

Comment:
I think you should keep per message logging to a trace/debug level. Possibly not even debug.